.github/workflows/pull_requests.yml

```diff
@@ -27,4 +27,33 @@ jobs:
 
             Also take a look at our [FAQ](https://github.com/terraform-providers/terraform-provider-aws/blob/main/docs/FAQ.md) which details how we prioritize Pull Requests for inclusion.
             
-            Thanks again, and welcome to the community! :smiley:         
+            Thanks again, and welcome to the community! :smiley:
+   ChangelogChecks:
+    # If there  a `pr/no-changelog` label we ignore this check
+    if: "${{!contains(github.event.pull_request.labels.*.name, 'pr/no-changelog')}}"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 # by default the checkout action doesn't checkout all branches
+      - name: Check for changelog entry in diff
+        run: |
+          # check if there is a diff in the .changelog directory
+          changelog_files=$(git --no-pager diff --name-only HEAD "$(git merge-base HEAD "origin/${{ github.event.pull_request.base.ref }}")" -- .changelog/)
```

Comment:
I believe `dorny/paths-filter` can be used to replace this lookup logic, see also `changelog.yml` 👍